src/main/java/com/checkmarx/flow/config/ScmConfigOverrider.java

```diff
@@ -41,20 +41,29 @@ private String getScmOverriddenConfig(RepoProperties properties, ScmConfigParams
             switch (configParams) {
                 case TOKEN:
                     log.info("Overriding token for SCM instance key: {}", key);
-                    value = optionalInstanceKey.getToken();
+                    value = validateOptionalInstanceValue(configParams, key, optionalInstanceKey.getToken());
                     break;
                 case WEBHOOK_TOKEN:
                     log.info("Overriding webhook-token for SCM instance key: {}", key);
-                    value = optionalInstanceKey.getWebhookToken();
+                    value = validateOptionalInstanceValue(configParams, key, optionalInstanceKey.getWebhookToken());
                     break;
                 case API_URL:
                     log.info("Overriding api-url for SCM instance key: {}", key);
-                    value = optionalInstanceKey.getApiUrl();
+                    value = validateOptionalInstanceValue(configParams, key, optionalInstanceKey.getApiUrl());
                     break;
                 default:
                     throw new MachinaRuntimeException("Scm key: " + key + "is not supported");
             }
         }
         return value;
     }
+
+    private String validateOptionalInstanceValue(ScmConfigParams configParams, String key, String value) {
+        if (Optional.ofNullable(value).isPresent()) {
```

Comment:
Why not just "if (value != null)"?

Reply:
You are right. I'm not using here any special advantages of Optional.